### PR TITLE
Misc updates to codeowners

### PR DESCRIPTION
**Pull request by Zimmi48**
**Kind:** meta.

The main change is the new @coq/parsing-maintainers team (for now consisting of Emilio, Hugo and Pierre-Marie). This follows an idea emitted by @ejgallego and this fixes an absence of code owners for some recently added files.

### CODE REVIEW

**Review by maximedenes — APPROVED** [review]

**Review by ejgallego — APPROVED** [review]
I am OK but in the long term `dune` files should be maintained by the maintainer of each component. with the dune-maintainers as secondary. They are not so hard to understand.

**Comment by Zimmi48** [comment]
Longer term, agreed. This is meant for the transition period.

File: .github/CODEOWNERS
Comment (ejgallego) [inline .github/CODEOWNERS]: A suggestion for a small tweak from PR #9024 :
```
/parsing/         @coq/parsing-maintainers
# Secondary maintainer @mattam82
/gramlib/         @coq/parsing-maintainers
# Secondary maintainer @nobody
/vernac/metasyntax.* @coq/parsing-maintainers
# Secondary maintainer @mattam82 @maximedenes
```

**Comment by ejgallego** [comment]
What is holding this PR ? This kind of delays are confusing as witnessed by #9024

**Comment by maximedenes** [comment]
> What is holding this PR ?

I forgot to assign myself. Feel free to assign me in such cases.

**Comment by maximedenes** [comment]
@ejgallego should I wait for your change or merge as is?

**Comment by Zimmi48** [comment]
I'll implement the recommended changes now.

**Comment by Zimmi48** [comment]
I pushed the additional line on `/vernac/metasyntax.*`
Adding secondary maintainers does not make sense in the case of a code owner team (even more so if the secondary are Matthieu and Maxime who are both not very responsive code owners :wink:).

**Comment by ejgallego** [comment]
Ping @maximedenes , this is indeed important as we are actively working on `gramlib`

**Comment by ejgallego** [comment]
Merci !

**Pull request merged**
